pkg/apis/core/validation/validation.go

```diff
@@ -1262,6 +1271,21 @@ func validateRBDPersistentVolumeSource(rbd *core.RBDPersistentVolumeSource, fldP
 	if len(rbd.RBDImage) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("image"), ""))
 	}
+	if rbd.SecretRef != nil {
+		if len(rbd.SecretRef.Name) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "name"), ""))
+		} else {
+			for _, msg := range ValidateSecretName(rbd.SecretRef.Name, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("secretRef", "name"), rbd.SecretRef.Name, msg))
+			}
+		}
+		if len(rbd.SecretRef.Namespace) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "namespace"), ""))
+		} else {
+			allErrs = append(allErrs, ValidateDNS1123Label(rbd.SecretRef.Namespace, fldPath.Child("secretRef", "namespace"))...)
+		}
+	}
```

Comment:
Please create a common function for `SecretRef` validation instead of copying it around.

Reply:
I was actually thinking of doing it.
I had one question though, Should this be inside the apimachinery package? As SecretReference object is nothing specific to volumes?

Reply:
I think it should be here, in validation.go, because `SecretReference` is declared in `pkg/apis/core` too

Reply:
Added [ValidateSecretReference](https://github.com/kubernetes/kubernetes/pull/103024/files#diff-c713e8919642d873fdf48fe8fb6d43e5cb2f53fd601066ff53580ea655948f0dR292) function